tests/unit_tests/blas/batch/CMakeLists.txt

```diff
@@ -29,7 +36,7 @@ if(BUILD_SHARED_LIBS)
       PUBLIC ${PROJECT_SOURCE_DIR}/include
       PUBLIC ${PROJECT_SOURCE_DIR}/deps/googletest/include
       PUBLIC ${CMAKE_BINARY_DIR}/bin
-      PUBLIC ${CBLAS_INCLUDE}
+      $<$<BOOL:${CBLAS_FOUND}>:${CBLAS_INCLUDE}>
```

Comment:
I don't really follow this syntax, but are we changing this argument from `PUBLIC` to non-public?

Reply:
New logic helps us to drop `CBLAS_INCLUDE` from the compilation line when CBLAS header file is not found. Otherwise it might appear in the compilation line as `-ICBLAS_INCLUDE_not_found` and cause test compilation issue. 

As for `PUBLIC` keyword since each BLAS test is a final binary, not library, we don't need to mark this dependency as needed to be propagated to other targets that depend on this target, because we don't have such targets.